pkg/controllers/rollout/controller.go

```diff
@@ -623,7 +623,10 @@ func isBindingReady(binding *fleetv1beta1.ClusterResourceBinding, readyTimeCutOf
 	// find the latest applied condition that has the same generation as the binding
 	availableCondition := binding.GetCondition(string(fleetv1beta1.ResourceBindingAvailable))
 	if condition.IsConditionStatusTrue(availableCondition, binding.GetGeneration()) {
-		if availableCondition.Reason != work.WorkNotTrackableReason {
+		// TO-DO (chenyu1): currently it checks for both the new and the old reason
+		// (as set previously by the work generator) to avoid compatibility issues.
+		// the check for the old reason can be removed once the rollout completes successfully.
+		if availableCondition.Reason != condition.WorkNotAvailabilityTrackableReason && availableCondition.Reason != workapplier.WorkNotAllManifestsTrackableReason {
```

Comment:
Reason is just a string, why change it?

Reply:
Hi Ryan! There was a minor consistency issue that triggered that change: for the work generator, it populates all the reason strings using values in similar formats (`AllWorks...` or `NotAllWorks...`) from the `condition package, except for the not trackable `Available` condition, which uses a string exported from the (old) work applier, which reads a bit weird.

Reply:
This also requires importing of the work applier package across multiple places in the code; in the long run I am guessing maybe we should put all reason strings in the same place?